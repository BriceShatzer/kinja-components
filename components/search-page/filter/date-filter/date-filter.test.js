import * as React from 'react';
import { shallow, mount } from 'enzyme';

import DateFilter from './';


describe('<DateFilter />', () => {
	const onChange = jest.fn();
	const todayTimemillisStub = 1559907522917; // 06/07/2019

	const stubElement = () => (
		<DateFilter
			defaultChecked="Show all stories"
			onChange={onChange}
			currentTimemillis={todayTimemillisStub}
		/>
	);

	beforeAll(() => {
		jest.spyOn(Date, 'now').mockImplementation(() => todayTimemillisStub);
	});

	it('should render properly', () => {
		const wrapper = shallow(stubElement());
		expect(wrapper).toMatchSnapshot();
	});

	it('should call onChange', () => {
		const wrapper = mount(stubElement());
		const radio = wrapper.find('Radio').at(1);
		radio.simulate('click');

		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalledWith({
			checked: 'Last 24 hours',
			dateStart: 1559907522917 - 86400000,
			dateEnd: null
		});
		onChange.mockClear();
	});

	it('should call onChange with customDate', () => {
		const wrapper = mount(stubElement());
		wrapper.instance().onCustomDateChange({ day: 2, month: 4, year: 2016 }, 'to');

		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalledWith({
			checked: 'Show all stories',
			dateStart: 915145200000,
			dateEnd: 1459569600000
		});
	});
});