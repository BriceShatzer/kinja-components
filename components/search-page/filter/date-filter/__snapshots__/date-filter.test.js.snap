// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateFilter /> should render properly 1`] = `
<EnsureDefaultTheme>
  <date-filter__Container>
    <Toggle(CollapsibleBox)
      isStatic={true}
      title="Filter by date"
    >
      <RadioGroup
        name="datefilter"
        onChange={[Function]}
      >
        <Radio
          checked={true}
          label="Show all stories"
          value="Show all stories"
        />
        <Radio
          checked={false}
          label="Last 24 hours"
          value="Last 24 hours"
        />
        <Radio
          checked={false}
          label="Last 48 hours"
          value="Last 48 hours"
        />
        <Radio
          checked={false}
          label="Past week"
          value="Past week"
        />
        <Radio
          checked={false}
          label="Past month"
          value="Past month"
        />
        <Radio
          checked={false}
          label="Specify date range"
          value="Specify date range"
        />
      </RadioGroup>
    </Toggle(CollapsibleBox)>
  </date-filter__Container>
</EnsureDefaultTheme>
`;
